companion-api/backend/client.go

```diff
@@ -0,0 +1,12 @@
+package backend
+
+import (
+	"github.com/mcluseau/autorizo/auth"
+)
+
+// Client is the interface for all backends clients
+type Client interface {
```

Comment:
this interface should probably be simpler and more flexible, like:
```golang
type UserData struct {
  PasswordHash string
  Claims auth.ExtraClaims
}

type Client interface {
  CreateUser(user string, data *UserData) error
  UpdateUser(user string, update func(data *UserData)) error
  DeleteUser(user string) error
}
```

Reply:
I use something like that:
```
// UserData is a simple user request with user id, paswordhash and claims
type UserData struct {
	PasswordHash string           `json:"password"`
	ExtraClaims  auth.ExtraClaims `json:"claims"`
}

// Client is the interface for all backends clients
type Client interface {
	CreateUser(id string, user *UserData) error
	UpdateUser(id string, userToUpdate *UserData, update func(oldUser, newUser *UserData)) error
	DeleteUser(id string) error
}
```

Reply:
lots of parameters for `UpdateUser`. The idea of `func update(...)` is that it's given the current UserData and changes it. So I don't see (yet) why it needs `userToUpdate` and 2 parameters.